src/components/shared/uicomponents/CustomerCommissionList.js

```diff
@@ -0,0 +1,36 @@
+import {DOMAIN_URL} from '../../../utils/Constants';
+
+import React from 'react';
+import axios from 'axios';
+
+export default class CustomerCommissionList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: true,
+            customerCommissions: [],
+        };
+    }
+
+    componentDidMount() {
+        axios.get(`${DOMAIN_URL}/api/v1/user/${this.props.userId}/customer-commissions`)
+        .then((response) => {
+            if (response.status === 200 || response.status === 304) {
+                this.setState({ customerCommissions: response.data.commissions });
+            }
+            this.setState({loading: false});
+        }).catch((error) => {
+            console.log(error);
+        });
+    }
+
+    render() {
+        return (
+            <ul>
+                {!(this.state.loading) && this.state.customerCommissions.map((commission) => {
```

Comment:
Although in Javascript `true && anything` returns `anything`, i dont believe its the correct way to do things. What should it should when `loading` is true then?

My suggestion would be something like this

```
<ul>
    {this.state.loading ? <li>Loading</li> : this.state.customerCommissions.map(...)}
</ul>
```

Reply:
ah that makes sense. didn't know about this syntax. I ran into some problems when trying to stick an if-else block in the <ul> but maybe that's just because i made syntax errors.

Reply:
you dont have to stick with teh short-hand syntax, if you;d like, use a proper if-else block